### PR TITLE
Move VectorFunctionReaction and other minor tweaks

### PR DESCRIPTION
**Pull request by nmnobre**
## motivation
VectorFunctionReaction could be used outside of the EM module, it is a very general functional-form object.
This will help framework and other modules developers create tests for vector equations

## Design
Move the source,include and doco for that file to the framework

## Impact
Easier development

Spin-off from #26105 with everything that I thought doesn't really belong there.
Moving `VectorFunctionReaction` will decrease its coverage momentarily, but that's only until we merge #26105 which will use that kernel in some framework tests.

### CODE REVIEW

**Review by cticenhour — APPROVED** [review]
LGTM! I have asked about the apptainer failure. It shouldn't be related to this, though I am not invalidating yet, until @loganharbour had a chance to look at it.

File: framework/src/kernels/VectorFunctionReaction.C [hunk 1]
```diff
@@ -9,7 +9,7 @@
 
 #include "VectorFunctionReaction.h"
 
-registerMooseObject("ElectromagneticsApp", VectorFunctionReaction);
+registerMooseObject("MooseApp", VectorFunctionReaction);
```
Comment (GiudGiud) [inline framework/src/kernels/VectorFunctionReaction.C:12]: for future reference, all these XXFunctionYY objects are easily outmatched by XXFunctorYY objects that can support functions, variables, functor material properties and even postprocessors. Scalar variable support soon

Comment (GiudGiud) [reply framework/src/kernels/VectorFunctionReaction.C:12]: no need to change anything here

Comment (cticenhour) [reply framework/src/kernels/VectorFunctionReaction.C:12]: This is a good point. Moving some of the EM related objects to a functor configuration is still something on my roadmap from when we first merged the module in. I'll get to it at some point!

Comment (lindsayad) [reply framework/src/kernels/VectorFunctionReaction.C:12]: Outmatched in flexibility, not in performance

**Comment by cticenhour** [comment]
@nmnobre I see some pushes since my last approval (which still stands - thanks for repackaging this into a smaller form from the RT PR!), I want to ask if this is final and ready to be merged?

**Comment by nmnobre** [comment]
Oh yes, it's final.

I just took the opportunity to fix a small typo and then rebased this morning in the hope that the apptainer test would pass, and it did. 😊

**Pull request merged**